packages/node/src/generators/application/application.ts

```diff
@@ -292,8 +294,8 @@ function updateTsConfigOptions(tree: Tree, options: NormalizedSchema) {
 }
 
 export async function applicationGenerator(tree: Tree, schema: Schema) {
-  // Prompt for bundler webpack / esbuild
-  if (schema.framework) {
+  // Prompt for bundler webpack / esbuild if framework is provided without '--bundler'
+  if (schema.framework && (!schema.bundler || schema.bundler === 'none')) {
```

Comment:
As discussed, let's skip prompting the bundler and just use `esbuild` as the default. Users can read the generator options or our docs if they need to change it.